[OSDOCS-7580] information about autoscaler using CLI (#63942)
openshift/openshift-docs: modules/rosa-cluster-autoscaler-cli-interactive.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-cluster-autoscaling.adoc
// * osd_cluster_admin/osd-cluster-autoscaling.adoc

:_content-type: REFERENCE
[id="rosa-cluster-autoscale-cli-interactive_{context}"]
= Enable autoscaling during cluster creation by using the interactive mode with the ROSA CLI

You can use the installer interactive mode to set cluster-wide autoscaling behavior during cluster creation.

Interactive mode has the benefit of providing more information about the available configurable parameters. Interactive mode also does basic checks and preflight validations, meaning that if a provided value is invalid, then it asks for a valid input.

.Procedure

. During cluster creation, use the `--enable autoscaling` and `-interactive` parameters to enable cluster autoscaling.
+
[source,terminal]
----
$ rosa create cluster --cluster-name <cluster_name> --enable-autoscaling -interactive
----

When the following prompt appears, enter *y* to go through all available autoscaling options.

[source,terminal]
----

> ? Configure cluster-autoscaler (optional): [? for help] (y/N)

Reviewer: ```suggestion
? Configure cluster-autoscaler (optional): [? for help] (y/N) y <enter>
```

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-cluster-autoscaling.adoc
// * osd_cluster_admin/osd-cluster-autoscaling.adoc

:_content-type: REFERENCE
[id="rosa-cluster-autoscale-cli-interactive_{context}"]

> = Enable autoscaling using the interactive mode with the ROSA CLI

Reviewer 2: s/using/by using

> = Enable autoscaling using the interactive mode with the ROSA CLI

Reviewer 2: ```suggestion
= Enable autoscaling during cluster creation by using the interactive mode with the ROSA CLI
```

> When running inside an interactive shell, the interactive autoscaling mode displays. This method has the benefit of providing more information about the different configurable parameters. It also does basic checks and preflight validations, meaning if a provided value is invalid, then it asks again for a valid input.

Reviewer 2: ```suggestion
When running inside an interactive shell, the interactive autoscaling mode is displayed. This method has the benefit of providing more information about the available configurable parameters. It also does basic checks and preflight validations, meaning that if a provided value is invalid, then it asks for a valid input.
```

.Procedure

> . To force interactive mode when not using an interactive shell, run the following command:

Reviewer 2: The intro paragraph talks about using an interactive shell, yet the procedure describes not using an interactive shell. I this confusing.

Reviewer 2: Need `+` characters between each line.

> . To force interactive mode when not using an interactive shell, run the following command:

Reviewer 2: The other "Enabling autoscaling modules do not have an intro paragraph before the procedure.
```suggestion
```

> . To force interactive mode when not using an interactive shell, run the following command:

Reviewer 2: ```suggestion
. During cluster creation, use the `--enable autoscaling` and `-interactive` parameters to enable machine autoscaling.
```

> 

Reviewer 2: Add `+` character.

[source,terminal]
----
$ rosa create cluster --cluster-name <cluster_name> --enable-autoscaling -interactive
----

Or

[source,terminal]
----
$ rosa create cluster --cluster-name <cluster_name> --enable-autoscaling -i

> ----

Reviewer 2: Probably don't need both examples.

> You still must accept setting cluster autoscaler parameters using this method. When

Reviewer 2: ```suggestion
. You must manually set all the cluster autoscaler parameters. When the following prompt appears, enter *y* to go through all available autoscaling options.
```

[source,terminal]
----
? Configure cluster-autoscaler (optional): [? for help] (y/N)
----

> appears, enter *y* to go through all available autoscaling options.

Reviewer 2: ~~appears, enter *y* to go through all available autoscaling options.~~
